test/suite/test_durable_ts03.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-2019 MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+from helper import copy_wiredtiger_home
+import wiredtiger, wttest
+
+def timestamp_str(t):
+    return '%x' %t
+
+# test_durable_ts03.py
+#    Check that the updates with durable timestamp newer than the stable
+#    timestamp fill up the cache and leave it stuck.
+class test_durable_ts03(wttest.WiredTigerTestCase):
+    # Reduce the cache size to 10MB to get the stuck cache.
+    def conn_config(self):
+        return 'cache_size=50MB'
+
+    def test_durable_ts03(self):
+        # Create a table.
+        uri = 'table:test_durable_ts03'
+        nrows = 600000
+        self.session.create(uri, 'key_format=i,value_format=u')
+        value1 = "aaaaa" * 100
+        value2 = "bbbbb" * 100
+
+        # Load the data into the table.
+        session = self.conn.open_session()
+        cursor = session.open_cursor(uri, None)
+        for i in range(0, nrows):
+            cursor[i] = value1
+        cursor.close()
+
+        # Set the stable timestamp to checkpoint initial data set.
+        self.conn.set_timestamp('stable_timestamp=' + timestamp_str(100))
```

Comment:
It's quite unusual to switch from not using timestamps, to using timestamps - and also more difficult to reason about expected behavior. You should switch to setting an oldest/stable timestamp at the start. Then using a timestamp of 50 for all of the intermediate writes, then updating the oldest timestamp?